osgp/platform/osgp-adapter-kafka-distributionautomation/src/test/java/org/opensmartgridplatform/adapter/kafka/da/infra/kafka/out/MeterReadingProducerTest.java

```diff
@@ -0,0 +1,104 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opensmartgridplatform.adapter.kafka.da.application.config.KafkaProducerConfig;
+import org.opensmartgridplatform.adapter.kafka.da.application.mapping.DistributionAutomationMapper;
+import org.opensmartgridplatform.adapter.kafka.da.avro.IntervalBlock;
+import org.opensmartgridplatform.adapter.kafka.da.avro.MeterReading;
+import org.opensmartgridplatform.adapter.kafka.da.avro.UsagePoint;
+import org.opensmartgridplatform.adapter.kafka.da.avro.ValuesInterval;
+import org.opensmartgridplatform.domain.da.measurements.MeasurementReport;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.test.EmbeddedKafkaBroker;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.kafka.test.utils.KafkaTestUtils;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+@SpringJUnitConfig(KafkaProducerConfig.class)
+@TestPropertySource("classpath:osgp-adapter-kafka-distributionautomation-test.properties")
+@ExtendWith(MockitoExtension.class)
+@EmbeddedKafka(partitions = 1,
+        topics = { "${distributionautomation.kafka.producer.topic}" },
+        brokerProperties = { "listeners=PLAINTEXT://localhost:9092", "log.dirs=../kafka-logs/",
+                "auto.create.topics.enable=true" })
+class MeterReadingProducerTest {
+
+    @Value("${distributionautomation.kafka.producer.topic}")
+    private String topic;
+
+    @Autowired
+    private EmbeddedKafkaBroker embeddedKafka;
+
+    @Mock
+    private DistributionAutomationMapper mapper;
+
+    @Autowired
+    private KafkaTemplate<String, MeterReading> template;
+
+    private MeterReadingProducer producer;
+    private MeterReading message;
+
+    @BeforeEach
+    @SuppressWarnings("unchecked")
+    public void setup() {
+        this.message = this.createMessage();
+        when(this.mapper.map(any(MeasurementReport.class), any(Class.class))).thenReturn(this.message);
+        this.producer = new MeterReadingProducer(this.template, this.mapper);
+    }
+
+    @Test
+    void sendTest() {
+
+        // send a message to the kafka bus
```

Comment:
Sander uses Arrange / Act / Assert as parts in a test. I think it's a good approach to structure a test.

Reply:
for newbies it is